src/test/java/com/google/sps/DataServletTest.java

```diff
@@ -0,0 +1,229 @@
+    when(request.getParameter("CourseClass"))
+        .thenReturn(Course("Spring 2020", "Smith", 12, termKey));
+
+    newTermRating.addTermRating(request);
+
+    Entity termRatingQuery =
+        queryEntities("Term Rating", reviewer - id, request.getParameter("reviewer-id")).get(0);
+    Entity expectedTermRatingEntity = new Entity("Term Rating");
+    expectedTermRatingEntity.setProperty("comments-class", "I really hate this class.");
+    assertEquals(
+        expectedTermRatingEntity.getProperty("comments-class"),
+        termRatingQuery.getProperty("comments-class"));
+    expectedTermRatingEntity.setProperty("reviewer-id", "numberOneId");
+    assertEquals(
+        expectedTermRatingEntity.getProperty("reviewer-id"),
+        termRatingQuery.getProperty("reviewer-id"));
+    expectedTermRatingEntity.setProperty("perception-class", 1);
+    assertEquals(
+        expectedTermRatingEntity.getProperty("perception-class"),
+        termRatingQuery.getProperty("perception-class"));
+    expectedTermRatingEntity.setProperty("hours", 10);
+    assertEquals(
+        expectedTermRatingEntity.getProperty("hours"), termRatingQuery.getProperty("hours"));
+    expectedTermRatingEntity.setProperty("difficulty", 5);
+    assertEquals(
+        expectedTermRatingEntity.getProperty("difficulty"),
+        termRatingQuery.getProperty("difficulty"));
+    expectedTermRatingEntity.setProperty("comments-professor", "The professor was still okay.");
+    assertEquals(
+        expectedTermRatingEntity.getProperty("comments-professor"),
+        termRatingQuery.getProperty("comments-professor"));
+    expectedTermRatingEntity.setProperty("perception-professor", 3);
+    assertEquals(
+        expectedTermRatingEntity.getProperty("perception-professor"),
+        termRatingQuery.getProperty("perception-professor"));
+  }
+
+  @Test
+  public void AddingTermRatingWithSentimentAnalysis() {
+
+    DatastoreService db = DatastoreServiceFactory.getDatastoreService();
+    request = Mockito.mock(HttpServletRequest.class);
+
+    when(request.getParameter("class-input")).thenReturn("I do not like this.");
+    when(request.getParameter("rating-class")).thenReturn("4");
+    when(request.getParameter("hoursOfWork")).thenReturn("8");
+    when(request.getParameter("difficulty")).thenReturn("4");
+    when(request.getParameter("prof-input")).thenReturn("The professor was amazing.");
+    when(request.getParameter("rating-professor")).thenReturn("3");
+    when(request.getParameter("languages")).thenReturn("false");
+    when(request.getParameter("ID")).thenReturn("numberOneId");
+    Entity termEntity = new Entity("Term");
+    Key termKey = termEntity.getKey();
+    when(request.getParameter("CourseClass"))
+        .thenReturn(Course("Spring 2020", "Smith", 12, termKey));
+
+    newTermRating.addTermRating(request);
+
+    Entity termRatingQuery =
+        queryEntities("Term Rating", reviewer - id, request.getParameter("reviewer-id")).get(0);
```

Comment:
missing quotes around "reviewer - id" I think you copied this throughout so check all of them

Reply:
Sorry about this, you are completely right hahaha